packages/worker/src/spot-termination.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	changeMessageVisibility,
+	logger,
+} from '@guardian/transcription-service-backend-common';
+import { SQSClient } from '@aws-sdk/client-sqs';
+import { getCurrentReceiptHandle, setInterruptionTime } from './index';
+
+const CHECK_FREQUENCY = 10;
+
+export const checkSpotInterrupt = async (
+	client: SQSClient,
+	queueUrl: string,
+) => {
+	const url = 'http://169.254.169.254/latest/meta-data/spot/instance-action';
+	const result = await fetch(url);
```

Comment:
do we need to catch network errors? seems like we need to be careful with error handling here since if there's an exception we'll stop polling for metadata

Reply:
good question, `169.254.169.254` is a local IP address for the instance metadata service so should be fairly reliable but I will wrap in a try/finally to be sure